src/components/HistoricWorthChart.tsx

```diff
@@ -49,19 +50,29 @@ const startingDates: Record<ChartLength, Dayjs> = {
 const HistoricWorthChart = memo(function HistoricWorthChart({
   addressHashes,
   currency,
+  length = '1y',
   onDataPointHover,
-  length = '1y'
+  onChartLengthChange
 }: HistoricWorthChartProps) {
   const addresses = useAppSelector((s) => selectAddresses(s, addressHashes))
   const haveHistoricBalancesLoaded = useAppSelector(selectHaveHistoricBalancesLoaded)
   const { data: alphPriceHistory } = useGetHistoricalPriceQuery({ currency, days: 365 })
   const theme = useTheme()
 
+  const previousLength = useRef<ChartLength>()
+
   const [chartData, setChartData] = useState<DataPoint[]>([])
 
   const startingDate = startingDates[length].format('YYYY-MM-DD')
   const isDataAvailable = addresses.length !== 0 && haveHistoricBalancesLoaded && alphPriceHistory
 
+  useEffect(() => {
+    if (chartData.length > 0 && (!previousLength.current || previousLength.current !== length)) {
```

Comment:
I think that due to the way we are processing the chart data, we should check `> 1` instead. I think the chartData will always have 1 element.

Reply:
Unless of course your changes change that ;)